### ISSUE TITLE
iOS Version - App Store Connect Reported Issues

### ISSUE THREAD

**Issue by workerbee22** [Western-Health-Covid19-Collaboration/wh_covid19_app#196]
When we submit to App Store Connect (used for TestFlight and the App Store) Apple analyses the app files and reports on any issues.

As at v1.0.101 we have these issues. Currently neither of these is stopping the build nor the release of the app for testing on iOS:

![image](https://user-images.githubusercontent.com/5154664/78614613-555ea800-78b2-11ea-8f7c-17d0f7568df5.png)

**Comment by workerbee22** [comment]
**ITMS-90078: Missing Push Notification Entitlement**

Is a Flutter framework issue, that is being worked on but the consensus is to ignore it for now. See Flutter issue:  https://github.com/flutter/flutter/issues/9984

**Comment by workerbee22** [comment]
**ITMS-90381: Too many symbol files**

Crept in as a new issue with v1.0.101 when I locally updated brew, ruby and cocoapods to make sure we were using cocoapods v1.9.1. 

There are fixes for this that I'm investigating such as this and others: https://stackoverflow.com/questions/34313049/too-many-symbol-files-warning-when-submitting-app

This may be caused by our change in v1.0.101 to only support iOS 12.0 and above.

**Comment by yimingc2** [comment]
> **ITMS-90381: Too many symbol files**
> 
> Crept in as a new issue with v1.0.101 when I locally updated brew, ruby and cocoapods to make sure we were using cocoapods v1.9.1.
> 
> There are fixes for this that I'm investigating such as this and others: https://stackoverflow.com/questions/34313049/too-many-symbol-files-warning-when-submitting-app
> 
> This may be caused by our change in v1.0.101 to only support iOS 12.0 and above.

@workerbee22 , the reason of the warning is that iOS deployment target for the Pods is iOS8 which is lower than the project deployment target iOS12. So it would generate armv7 dSYM files but no binaries.
So the solution is add 
`config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '12.0'`
to the Podfile.

**Comment by workerbee22** [comment]
Awesome thanks @yimingc2 . I'm also looking at this SO article to confirm syntax etc: https://stackoverflow.com/questions/54704207/the-ios-simulator-deployment-targets-is-set-to-7-0-but-the-range-of-supported-d

So this is as per your solution above June and seems to be the best answer:

![image](https://user-images.githubusercontent.com/5154664/78949654-84258a00-7b0f-11ea-8fa8-7b442212ba35.png)

 Only will be `12.0` for us.

**Comment by yimingc2** [comment]
Yup, after I added the deploy target for pods, there is only arm64 and no
arm7 anymore so Apple should be happy with that. My change is just on my
branch. You can make the change for a PR as it’s one line. 😃

**Comment by workerbee22** [comment]
Yeah working on a PR for a couple of App Store Connect things, so I'll do this fix and another in that PR shortly 👍

**Comment by workerbee22** [comment]
Still looking at this one, previous PR failed to fix the issue. @yimingc2 lets discuss the entry for `config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '12.0'` because it sems to be in there and yet not fixed.

**Comment by workerbee22** [comment]
UI fix failed, so discussed with @yimingc2 the Podfile change by adding this line:

```
post_install do |installer|
  installer.pods_project.targets.each do |target|
    target.build_configurations.each do |config|
      config.build_settings['ENABLE_BITCODE'] = 'NO'
      config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '12.0'
    end
  end
end
```

**Comment by workerbee22** [comment]
The warning **ITMS-90078: Missing Push Notification Entitlement** remains, but as per Flutter issue. But will ignore this for now as we track the flutter framework issue on this [issue](https://github.com/flutter/flutter/issues/9984)

**Comment by workerbee22** [comment]
The **ITMS-90381: Too many symbol** files is resolved. 

Now only the warning **ITMS-90078: Missing Push Notification Entitlement** remains. We'll see if this gets resolved once we move to a new flutter version.

Flutter frame work issue is here: flutter/flutter#9984 but summary is ' ... turn on Push Notifications then the warning will go way ...'  🤣Not sure we should do that when we can ignore it for now with no adverse effects.

**Comment by workerbee22** [comment]
Will retest when move to Flutter stable v1.17 as part of [issue 293](https://github.com/Western-Health-Covid19-Collaboration/wh_covid19_app/issues/293)

**Comment by workerbee22** [comment]
Warning **ITMS-90078: Missing Push Notification Entitlement** remains.

Still an open issue after Flutter v1.17.1 as the new sta le release. Recommendation is still to ignore the warning:

https://github.com/flutter/flutter/issues/9984
